test/routing/VersionSpec.scala

```diff
@@ -18,23 +18,41 @@ package routing
 
 import play.api.http.HeaderNames.ACCEPT
 import play.api.test.FakeRequest
+import play.api.libs.json._
 import support.UnitSpec
 
 class VersionSpec extends UnitSpec {
 
-  "Versions" when {
-    "retrieved from a request header" must {
-      "return an error if the version is unsupported" in {
-        Versions.getFromRequest(FakeRequest().withHeaders((ACCEPT, "application/vnd.hmrc.5.0+json"))) shouldBe Left(VersionNotFound)
+  "VersionReads" when {
+    "reading a valid version string" should {
+      "return Version3" in {
+        JsString(Version3.name).validate[Version](Version.VersionReads) shouldBe JsSuccess(Version3)
```

Comment:
need to assign to "result" along with the type, as "shouldBe" isn't type-safe

Reply:
(also it makes it a bit clearer)

Reply:
Actually can da whole thing be something more like this (from sa-individual-details-api, appearing in a PR soon):

```
    "serialized to Json" must {
      "return the expected Json output" in {
        val version: Version = Version1
        val expected         = Json.parse(""" "1.0" """)
        val result           = Json.toJson(version)
        result shouldBe expected
      }
    }

    "deserialized from a Json array" must {
      "return the expected version enum" in {
        val result = Json.parse(""" [ "1.0" ]""").as[Seq[Version]]
        result shouldBe List(Version1)
      }

      "read from Json containing an invalid version" in {
        val result: JsResult[Seq[Version]] = Json.parse(""" [ "99.0" ]""").validate[Seq[Version]]
        result shouldBe JsError(JsPath(0), "Unrecognised version")
      }
    }
```

(also includes the Writes, just for completeness)